### DISCUSSION TITLE
Why does the supabase cli generate types always change the Date value to a string?

### DISCUSSION THREAD

**Discussion by Sisableng** [supabase/supabase#27556] [Questions] [answered]
why does the supabase cli generate types always change the Date value to a string? even the value type for the colum `example: updated_at` is `timestampz`, generate types changes it to a string in **_database.types.ts_** which should be a _**Date**_ type. Can someone explain or have a good solution?

Column type
![Screenshot (19)](https://github.com/supabase/supabase/assets/58748827/4ef71591-5bda-4465-819a-e6ff3f45ef09)

Generate types
```typescript
posts: {
  Row: {
    created_at: string
    updated_at: string | null
  }
  Insert: {
    created_at?: string
    updated_at?: string | null
  }
  Update: {
    created_at?: string
    updated_at?: string | null
  }
}
```

**Comment by GaryAustin1** [comment, marked as answer]
Discussion on the reason:
https://github.com/supabase/postgrest-js/issues/201

**Comment by cemo** [comment]
Any workaround for this?
